### PR TITLE
feat: add cliffy + node support

### PR DESCRIPTION
**Pull request by ryanleecode**
adds cliffy and support for running eg-ts on with node

### CODE REVIEW

File: cli/deno.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
+import { Buffer, readLines } from "../deps/std/io.ts"
+import { readerFromStreamReader, writeAll } from "../deps/std/streams.ts"
+
+export interface DenoRunFlags {
+  reload?: string | undefined
```
Comment (harrysolovay) [inline cli/deno.ts:5]:
```suggestion
  reload?: string
```

File: cli/browser.ts [hunk 2]
```diff
@@ -0,0 +1,27 @@
+import { unimplemented } from "../deps/std/assert.ts"
+import { Buffer } from "../deps/std/io.ts"
+
+export interface BrowserRunFlags {
+  browser?: string | undefined
```
Comment (harrysolovay) [inline cli/browser.ts:5]:
```suggestion
  browser?: string
```

File: cli/node.ts [hunk 3]
```diff
@@ -0,0 +1,24 @@
+import { Buffer, readLines } from "../deps/std/io.ts"
+import { readerFromStreamReader, writeAll } from "../deps/std/streams.ts"
+
+export interface NodeRunFlags {
```
Comment (harrysolovay) [inline cli/node.ts:4]: Can we get rid?

File: cli/node.ts [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+}
+
+export default function run({}: NodeRunFlags) {
```
Comment (harrysolovay) [inline cli/node.ts:7]: What can we share between the Deno and Node.JS `run`s?

Comment (harrysolovay) [reply cli/node.ts:7]: Seemingly everything beneath the spawning of the process can be reused between the two?

**Review by harrysolovay — CHANGES_REQUESTED** [review]

File: cli/process.ts [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+    const process = new Deno.Command("node", {
+      args: [pathname],
+      stdout: "piped",
+      stderr: "piped",
+    }).spawn()
+    return runProcess(process, logs)
+  }
+}
+
+async function runProcess(process: Deno.ChildProcess, logs: Buffer) {
```
Comment (harrysolovay) [inline cli/process.ts:31]: I don't love this name given that it accepts the already-run/spawned process. Any other name ideas?

Comment (ryanleecode) [reply cli/process.ts:31]: pipeLogs?

Comment (ryanleecode) [reply cli/process.ts]: went with collectResults

**Review by harrysolovay — APPROVED** [review]
LGTM

**Pull request merged**